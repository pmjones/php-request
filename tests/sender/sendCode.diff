001+ Fatal error: Uncaught Error: Call to undefined method SapiResponse::setCode() in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCode.php:3
002+ Stack trace:
003+ #0 {main}
001- array(0) {
002- }
003- int(400)
004+   thrown in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCode.php on line 3