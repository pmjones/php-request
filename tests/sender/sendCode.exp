array(0) {
}
int(400)