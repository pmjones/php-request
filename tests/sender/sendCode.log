
---- EXPECTED OUTPUT
array(0) {
}
int(400)
---- ACTUAL OUTPUT
Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(SapiResponse))
#1 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCode.php(4): SapiResponseSender->send(Object(SapiResponse))
#2 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20
---- FAILED
