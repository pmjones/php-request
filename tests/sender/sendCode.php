<?php
$response = new SapiResponse();
$response->setCode(400);
(new SapiResponseSender())->send($response);
var_dump(headers_list());
var_dump(http_response_code());
