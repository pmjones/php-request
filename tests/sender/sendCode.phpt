--TEST--
SapiResponseSender::sendStatus
--CGI--
--INI--
expose_php=0
--FILE--
<?php
$response = new SapiResponse();
$response->setCode(400);
(new SapiResponseSender())->send($response);
var_dump(headers_list());
var_dump(http_response_code());
--EXPECTHEADERS--
--EXPECT--
array(0) {
}
int(400)
