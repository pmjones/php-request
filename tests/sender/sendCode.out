Fatal error: Uncaught Error: Call to undefined method SapiResponse::setCode() in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCode.php:3
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCode.php on line 3